### DISCUSSION TITLE
Deleting old users not found.

### DISCUSSION THREAD

**Discussion by thabaum** [oqtane/oqtane.framework#4768] [Q&A]
I have 2 users created a while back for testing.  They I dont believe have been deleted, however they are still in this installation.  The `IsDeleted` field is set to `False` so they had never been deleted from a control panel.

I see them in the `Users` table and the `AspNetUsers` table along with roles set for them in the `UserRoles` table along with some settings associated with them in the `Settings` table.

I am wondering why I do not see these users in the User Management control panel so I can delete them, and if there is a way to delete them safely manually?  Should these users actually go to the recycle bin?  Is there a purge deleted users scheduled task?

Just trying to think of how to handle deleted users or rogue users like these.  I may have created them with another site that was deleted in the past while testing to make these not found in User Management in the rest of sites in this single database multi-site instance of Oqtane.

**Comment by sbwalker** [comment]
Users must be in the Registered Users role for a specific site for them to appear in User Management for that site.

**Reply by thabaum** [reply]
Gotcha, I added them to `RoleID` in the `UserRoles` and they now appear for that site with that particular role id.  Thanks!

I don't see a way to un-register them or a way to "disable" the user.  Just a way to delete them.  

When you delete, it just removes them from the 'UserRole' table.  The "old" user is still in the `User` table and the "IsDeleted" is still set to 'False' in the 'User' table.

So I take it I need to re-add them to the same site they had been created with since I created a user with that same site I added them to the RoleID and the user I created to test was in fact deleted from the database.

The `AspNetUsers` table the old and new user both were removed from.  But the old user again is still in the `Users` table.

I would think they should go to the recycle bin with a soft delete first to deactivate the accounts, then if you empty recycle bin it removes them entirely possibly?

**Reply by thabaum** [reply]
I think at this point I can manually remove the entries from the User table.  I think they are still there because they have settings in the Settings table that maybe the issue?  CP-pane = Admin and CP-category = Admin and CP-editmode = -1 For Entity name User wtih the entity ID set to the same as this user?

**Comment by sbwalker** [comment]
If you have multiple sites in your installation and they are managed in the same database, then the User record is shared across the sites. So a user could be assigned to roles in Site1 but not be assigned to any roles in Site2. The Recycle Bin is not designed for Users.

**Reply by thabaum** [reply]
OK, I understand that the user roles ID's are specific to the site, multi-site sharing same database.  

I added the user to a role for a site, it appeared, I deleted the user, the user was deleted for AspNetUsers table but not the Users table and the users Settings are still in the database as well.  

This user cannot log in so no way to utilize any roles on any site in that database.  The site is empty, no pages, no modules as it was used for testing. 

I have added the user to role from the original site as well and tried with no success.  I think since the user has settings in the setting table I believe it wont fully delete it.

It should be safe to delete these user setttings from the Setting table that relate to the User ID, correct?  I can then attempt the delete user again and see if it removes it from the database as it does with other users created.

**Reply by thabaum** [reply]
The user was deleted from the UserRoles when deleted FYI, just left behind Setting and User database entries.

**Reply by sbwalker** [reply]
there is no referential integrity (FK) between User and Settings (this is by design). So I dont think the Settings are what caused it to not be deleted.

**Reply by thabaum** [reply]
My guess is maybe I had another site they had been created with, the site was deleted and these users left behind and not removed along with settings for them as they may have been added to admin roles prior to deleting the site.

**Reply by thabaum** [reply]
maybe not admin role, but permissions to edit the admin page for example.
